platform/android/export/export_plugin.cpp

```diff
@@ -2201,43 +2203,68 @@ bool EditorExportPlatformAndroid::can_export(const Ref<EditorExportPreset> &p_pr
 		if (hand_tracking > XR_HAND_TRACKING_NONE) {
 			valid = false;
-			err += TTR("\"Hand Tracking\" is only valid when \"Xr Mode\" is \"OpenXR\".");
+			err += TTR("\"Hand Tracking\" is only valid when \"XR Mode\" is \"OpenXR\".");
 			err += "\n";
 		}
 
 		if (passthrough_mode > XR_PASSTHROUGH_NONE) {
 			valid = false;
-			err += TTR("\"Passthrough\" is only valid when \"Xr Mode\" is \"OpenXR\".");
+			err += TTR("\"Passthrough\" is only valid when \"XR Mode\" is \"OpenXR\".");
 			err += "\n";
 		}
 	}
 
-	if (int(p_preset->get("custom_template/export_format")) == EXPORT_FORMAT_AAB &&
+	if (int(p_preset->get("custom_build/export_format")) == EXPORT_FORMAT_AAB &&
 			!custom_build_enabled) {
 		valid = false;
 		err += TTR("\"Export AAB\" is only valid when \"Use Custom Build\" is enabled.");
 		err += "\n";
 	}
 
-	// Check the min sdk version
-	int min_sdk_version = p_preset->get("version/min_sdk");
-	if (min_sdk_version != DEFAULT_MIN_SDK_VERSION && !custom_build_enabled) {
-		valid = false;
-		err += TTR("Changing the \"Min Sdk\" is only valid when \"Use Custom Build\" is enabled.");
-		err += "\n";
+	// Check the min sdk version.
+	String min_sdk_str = p_preset->get("custom_build/min_sdk");
+	int min_sdk_int = DEFAULT_MIN_SDK_VERSION;
+	if (!min_sdk_str.is_empty()) { // Empty means no override, nothing to do.
+		if (!custom_build_enabled) {
+			valid = false;
+			err += TTR("\"Min SDK\" can only be overridden when \"Use Custom Build\" is enabled.");
+			err += "\n";
+		}
+		if (!min_sdk_str.is_valid_int()) {
+			valid = false;
+			err += vformat(TTR("\"Min SDK\" should be a valid integer, but got \"%s\" which is invalid."), min_sdk_str);
+			err += "\n";
+		} else {
+			min_sdk_int = min_sdk_str.to_int();
+			if (min_sdk_int < DEFAULT_MIN_SDK_VERSION) {
+				valid = false;
+				err += vformat(TTR("\"Min SDK\" cannot be lower than %d, which is the version needed by the Godot library."), DEFAULT_MIN_SDK_VERSION);
+				err += "\n";
+			}
+		}
 	}
 
-	// Check the target sdk version
-	int target_sdk_version = p_preset->get("version/target_sdk");
-	if (target_sdk_version != DEFAULT_TARGET_SDK_VERSION && !custom_build_enabled) {
-		valid = false;
-		err += TTR("Changing the \"Target Sdk\" is only valid when \"Use Custom Build\" is enabled.");
-		err += "\n";
+	// Check the target sdk version.
+	String target_sdk_str = p_preset->get("custom_build/target_sdk");
+	int target_sdk_int = DEFAULT_TARGET_SDK_VERSION;
+	if (!target_sdk_str.is_empty()) { // Empty means no override, nothing to do.
+		if (!custom_build_enabled) {
+			valid = false;
+			err += TTR("\"Target SDK\" can only be overridden when \"Use Custom Build\" is enabled.");
+			err += "\n";
+		}
+		if (!target_sdk_str.is_valid_int()) {
+			valid = false;
+			err += vformat(TTR("\"Target SDK\" should be a valid integer, but got \"%s\" which is invalid."), target_sdk_str);
+			err += "\n";
+		} else {
+			target_sdk_int = target_sdk_str.to_int();
+		}
```

Comment:
I agree with not capping the target sdk (surprisingly, gradle doesn't complain when you use insanely high values of the target sdk), but we should include a warning to let users know that the engine hasn't been tested with the specified target sdk value and may be instable.

This will let advanced users test new versions of the target sdk and report issues if they encounter any (speeding up our upgrade process), while regular users will be nudged back toward the safer (tested) versions.

Reply:
> as it's typically used to
support newer SDKs faster than we do with official builds

Note that customizing the target sdk build is also done when targeting other stores that don't have the play store requirements (e.g: Oculus store, Amazon Android store).

Reply:
I agree, but we don't really have a good way to show such warnings to the users without being obnoxious if they actually know what they're doing.

This validation code can only add blocking errors, and while we could do a `WARN_PRINT`, it's easy to miss and will always be printed whenever they open the export dialog. Similarly, showing an info dialog in the editor would also happen every time.

We'd have to implement a custom dialog with a "Don't show this again." checkbox which saves to editor metadata for future occurrences (and somehow has logic to get reset if the conditions for the warnings change). That's significant work and IMO outside the scope of this PR.

Reply:
@akien-mga we already support printing warnings by updating the `err` string but setting `valid` to true.
See https://github.com/godotengine/godot/blob/master/editor/project_export.cpp#L277-L287

Reply:
Thanks, adding a warning as we discussed.